### PR TITLE
Migrate last remaining tests from Travis

### PR DESCRIPTION
**Pull request by dra27**
This PR completes the migration of checks from Travis to GitHub actions.

There are three tests performed:
- Ensure that `make alldepend` is a no-op
- Manual tests
- Labelled/Unlabelled module docs in sync

### CODE REVIEW

File: .github/workflows/hygiene.yml [hunk 1]
```diff
@@ -68,3 +68,24 @@ jobs:
           && (startsWith(github.event.ref, 'refs/heads/4.')
              || github.event.ref == 'refs/heads/trunk')
           && always()
+
+      - name: Check that labelled/unlabelled .mli files are in sync
+        run: tools/ci/actions/check-labelled-interfaces.sh
+        if: always()
+
+        # If this step fails, it leaves behind a file the presence of which can
```
Comment (gasche) [inline .github/workflows/hygiene.yml]: naming the file would help reading the `if:` clauses below

**Review by gasche — APPROVED** [review]

File: .github/workflows/hygiene.yml (hunk 1 above)
Comment (dra27) [reply .github/workflows/hygiene.yml]: Done!

**Comment by dra27** [comment]
While checking that the tests break, I discovered that the `check-stdlib` test in the manual has been broken for some time - I've pushed a commit to fix it!

**Comment by dra27** [comment]
cc @Octachron for just glancing at https://github.com/ocaml/ocaml/pull/10214/commits/26286c2a6da742a542021bacd45bd619a268e3d4 as the correct fix for this

**Comment by gasche** [comment]
Two more comments on those Github Actions:
- "Checks" has some actions that are instantaneous, and two that depend on a partial compiler build; on the last run the first group runs in <10s and the second group runs in ~30s. Would it make sense to split in two independent actions to reduce the feedback loop on the first group?
- How are "allowed failures" displayed and handled? On Travis we would often forget completely about allowed-failures tests failing (and I think your `check-stdlib` thing demonstrates), because one has to click a button to see their result explicitly. What happens if one of those checks fails here, is it visible directly from the PR results?

**Comment by dra27** [comment]
They can indeed be split: the pro is the obvious increase in parallelism. The con is the obvious increase in parallelism! This way, each PR consumes one of our parallel job allowances (I think we can have either 10 or 15 Linux jobs running at once?!). The more serious implication - which I know (rightly) bothers @xavierleroy is that the split will take an extra line in the "checks" list on the PR... at present, there's one master "Hygiene / Checks" line.

I don't have a strong opinion - I think it's marginally better to be using one job per PR on these checks.

**Comment by dra27** [comment]
We stopped using allowed-failures on Travis (because no-one checked them) and we haven't started using them in GitHub Actions. I think all the previous UI problems still apply.

The bug I've fixed in `check-stdlib` is because the test was broken - I found that by intentionally breaking it on my local fork and not seeing that breakage reported by CI. That's not related to allowed-failures.

All Hygiene checks which can run are run, regardless of failures (otherwise you'd only be told about a stray space at EOL after adding a Changes entry!). The overall workflow fails if _any_ step fails. So if any Hygiene test fails you get a red cross - finding out which test it is is the first part of reading the log (the step will have a red cross). The UI improvement over Travis is that you see Build failures completely separately from Hygiene, so you can choose to ignore the failure of that job early on in a PR without losing information you really ought to care about - the key thing is that everything tested in Hygiene has an easy and near-automatic resolution.

Steps are skipped if they are meaningless. At present those are:
- Changes entry for a push (or if the label has been applied to the PR)
- check-typo over the whole repo for a PR
- The alldepend and manual tests skip _if building the compiler failed_ (because they can't be run) - but note that the Hygiene will still fail, but presumably so will one or all of the Build jobs, because you broken OCaml! Pedantically, technically the manual tests only need configure to have succeeded, but even I considered having two file cookies for errors to be over-engineering 😉

**Comment by gasche** [comment]
Thanks for the explanations.

> I don't have a strong opinion - I think it's marginally better to be using one job per PR on these checks.

Let's marginally not do anything for now then.

**Comment by xavierleroy** [comment]
> The more serious implication - which I know (rightly) bothers @xavierleroy is that the split will take an extra line in the "checks" list on the PR... at present, there's one master "Hygiene / Checks" line.

Actually, I'm not bothered... or maybe bothered the other way (if it makes any sense!).  One thing I didn't like with Travis CI is that when it failed I had to click on its link to see which of the many tests had failed.  Then, for some reason, the back button of my browser would not work, or had to be pressed 5 to 7 times to get me back to the Github page.   So, seeing one line per check on the Github page (instead of a single "Hygiene/Checks" line) could be even better, hoping there are not so many lines that they fill the screen.  But that's really minor.

> This way, each PR consumes one of our parallel job allowances (I think we can have either 10 or 15 Linux jobs running at once?!).

Not exceeding our resource quota is important too...

**Comment by MisterDA** [comment]
The documentation at https://github.com/ocaml/ocaml/blob/trunk/HACKING.adoc#githubs-ci-travis-and-appveyor hasn't been updated to account for the removal of Travis.

**Pull request merged**